### PR TITLE
[Enhancement] Enforce const and immutability in cache management

### PR DESCRIPTION
**Pull request by sduzh**
Changed the shared_ptr in the tablet metadata to const to emphasize the immutability of the metadata after initial creation.
Adjusted some function calls and related logic to match these changes. This will prevent the accidental modification of these data, improving overall stability. 

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 43 (86.05%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/rowset/segment.h|0|1|00.00%|[170]|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|8|11|72.73%|[245, 252, 278]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|17|19|89.47%|[536, 765]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 159 / 180 (88.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[258, 259, 260, 266, 272, 274]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|22|28|78.57%|[347, 399, 408, 414, 549, 813]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|101|110|91.82%|[77, 151, 152, 197, 198, 199, 200, 201, 202]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 159 / 178 (89.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[258, 259, 260, 266, 272, 274]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|18|22|81.82%|[348, 400, 538, 603]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|101|110|91.82%|[77, 151, 152, 197, 198, 199, 200, 201, 202]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 160 / 176 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[258, 259, 260, 266, 272, 274]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|20|24|83.33%|[328, 380, 518, 583]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|99|105|94.29%|[145, 146, 163, 164, 181, 182]|
|:large_blue_circle:|src/storage/lake/schema_change.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

File: be/src/storage/lake/metacache.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
```
Comment (sduzh) [inline be/src/storage/lake/metacache.cpp:1]: All variables and methods in this file are moved from TabletManager, no newly added variable or method

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 170 / 180 (94.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[257, 258, 259, 265, 271, 273]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|20|24|83.33%|[328, 380, 518, 583]|
|:large_blue_circle:|src/storage/lake/schema_change.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|6|6|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|105|105|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 156 / 174 (89.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[258, 259, 260, 266, 272, 274]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|15|18|83.33%|[308, 455, 507]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|101|110|91.82%|[77, 151, 152, 197, 198, 199, 200, 201, 202]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 156 / 174 (89.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|11|17|64.71%|[258, 259, 260, 266, 272, 274]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|15|18|83.33%|[308, 455, 507]|
|:large_blue_circle:|src/storage/lake/metacache.cpp|101|110|91.82%|[77, 151, 152, 197, 198, 199, 200, 201, 202]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/lake/metacache.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 175 / 190 (92.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|be/src/storage/lake/meta_file.cpp|11|18|61.11%|[257, 258, 259, 265, 266, 271, 273]|
|:large_blue_circle:|be/src/storage/lake/tablet_manager.cpp|17|20|85.00%|[288, 435, 487]|
|:large_blue_circle:|be/src/storage/lake/metacache.cpp|113|118|95.76%|[104, 122, 140, 158, 176]|
|:large_blue_circle:|be/src/util/lru_cache.cpp|9|9|100.00%|[]|
|:large_blue_circle:|be/src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/tablet.cpp|6|6|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/update_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/tablet_manager.h|1|1|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/txn_log_applier.cpp|3|3|100.00%|[]|
|:large_blue_circle:|be/src/storage/rowset/segment.cpp|1|1|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/schema_change.cpp|1|1|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/transactions.cpp|4|4|100.00%|[]|
|:large_blue_circle:|be/src/storage/lake/metacache.h|1|1|100.00%|[]|
|:large_blue_circle:|be/src/storage/rowset/column_reader.cpp|1|1|100.00%|[]|

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**